### ISSUE TITLE
Distances somtimes get clipped behind solid objects

### ISSUE THREAD

**Issue by yonatan8070** [dune3d/dune3d#84]
Distances can get clipped if a solid object is in the way. For example, I made a cube, extruded it, and then extruded a new cube into it to make a hole, I then applied a distance constraint to one of the vertical lines of the hole, and the number can get partially or fully clipped behind the solid. Here are screenshots:

Text is partially clipped from here
![image](https://github.com/dune3d/dune3d/assets/45665471/2772a5b5-f87c-4b28-b8d8-4f38bc1ed313)

Text is entirely invisible from here
![image](https://github.com/dune3d/dune3d/assets/45665471/a261bc3e-4653-4ac1-9306-a670aa489907)

I know that the constraint can be moved, but until I figured that out, the default position for it was hidden from almost any direction
![image](https://github.com/dune3d/dune3d/assets/45665471/13421b53-ca1e-44e3-bc85-393411a16f32)

**Comment by carrotIndustries** [comment]
Yeah, this is low-key bothering me as well, but so far I couldn't come up with a good solution to fix this problem

In the meanwhile, the recommended workaround is to temporarily turn off the solid model and move the dimension somewhere it's not clipped.

**Comment by yonatan8070** [comment]
I don't know if this is possible with Dune3D's architecture. Maybe you could move all the constraints, points, dimensions etc. to their own rendering layer and overlay that on top of the solid graphics, and while rendering have a check for whether the object a constraint is attached to is visible, and only render the constraint if it is

**Comment by dwbrite** [comment]
Until a rendering solution can be figured out, you can use alt+s to disable solid geometry.
